src/tutorial/hello-world/README.md

```diff
@@ -20,7 +20,32 @@ Each part will look as follows for the string `"Hello, world!"`:
 0x40            48656c6c6f2c20776f726c642100000000000000000000000000000000000000 // Value "Hello, world!"
 ```
 
-Encoding dynamic values takes alot of work!! In order to return `"Hello, world!"` we must return 96 bytes!.
+Encoding dynamic values takes alot of work!! In order to return `"Hello, world!"` we must return 96 bytes!
+
+You might think that we can populate the memory in sequential order starting with the location (0x00), then length (0x20), and then the value (0x40), however, this would result in the value "Hello, world!" being left padded, so we would get: 
+```
+[Byte number]   [DATA]    
+0x00            0000000000000000000000000000000000000000000000000000000000000020 // The location of the "Hello, world!" data (dynamic type).
+0x20            000000000000000000000000000000000000000000000000000000000000000d // The length of "Hello, world!" in bytes
+0x40            0000000000000000000000000000000000000048656c6c6f2c20776f726c6421 // Wrong encoding!
+```
+And that's not what we need. Instead, we're going to leverage what we know about memory to produce the right result, and that is:
+- Memory is always expanded in 32 byte increments
+- We can store a value starting from any index
+- We want to store the location starting at 0x00 and the length starting at 0x20
```

Comment:
May be worth separating out this one onto a new line, the other 3 bullets are facts while this one is a circumstance.

Reply:
You're right, makes more sense to have it separated.